pkg/runtime/reconciler.go

```diff
@@ -37,11 +37,13 @@ import (
 // controller-runtime.Controller objects (each containing a single reconciler
 // object)s and sharing watch and informer queues across those controllers.
 type reconciler struct {
-	kc  client.Client
-	rmf acktypes.AWSResourceManagerFactory
-	rd  acktypes.AWSResourceDescriptor
-	log logr.Logger
-	cfg Config
+	kc           client.Client
+	rmf          acktypes.AWSResourceManagerFactory
+	rd           acktypes.AWSResourceDescriptor
+	log          logr.Logger
+	cfg          Config
+	nsaCache     *namespaceAnnotationsCache
```

Comment:
probably cleaner to have a `pkg/runtime/cache.Caches` object which itself contains all the different caches.

Something like this:

```go
type Caches struct {
    Namespaces *NamespaceCache
    Accounts *AccountCache
}
```

along with a `pkg/runtime/cache.New` function that returns an initialized `pkg/runtime/cache.Caches` object:

```go
func New(kc clientset.Interface, log logr,Log) Caches {
    return Caches{
        Namespaces: newNamespaceCache(kc. log),
        Accounts: newAccountCache(kc, log),
    }
}
```

and then the `reconciler` can use it like so:

```go
import (
...
    ackrtcache "github.com/aws/aws-controllers-k8s/pkg/runtime/cache"
...
type reconciler struct {
...
    cache ackrtcache.Caches
}
...
    if r.cache.Namespaces.GetByKey(...) {
...
```

Reply:
Great will do that!